### ISSUE TITLE
some thoughts about the landing page example

### ISSUE THREAD

**Issue by bertdv** [biaslab/RxInfer.jl#16]
The example on the landing page is great. Very clear code and it shows neatly how (1) model specification can be separated from inference; (2) inference can be automated. 

![Screenshot 2022-11-07 at 20 29 47](https://user-images.githubusercontent.com/3426064/200398090-7125c6c6-4a47-48ef-963d-c60558592cdb.png)

At the same time, here are some thoughts for possible improvements in the future:

### On `randomvar`
Do you need to specify `x = randomvar(n)`? What happens if you omit this specification? Why not just specify that `x` is a vector of `float64`s or omit it altogether and the rxinfer/model compiler infers from `x[i] ~ MvNormal( \mu = , \Sigma = )` that `x` is a `float64`?  If you say `x` is distributed as a Gaussian, then it must be a `float64`, unless specified otherwise (like a double float, or whatever you have in Julia). What do you really say by specifying that `x` is a randomvar? The probability distribution over `x` is not its type, but rather a statement of beliefs. (It's too big a discussion for this issue but the whole concept of a _random variable_ is very shady (if not plainly wrong) in a Bayesian framework). 

### On `datavar`
With`y = datavar`, you are really specifying that `y` is observed, but that should not be part of the knowledge base in the generative model specification. `y` is just a vector of `float64`'s, which can be inferred from the statement `y[i] = MvNormal(\mu= ,\Sigma= )`. I am not a fan of double type assignments (first by `y = datavar` and then by `y[i]=MvNormal()`), unless the type is not the default type that is associated with `MvNormal`. Another serious problem is that this model can not be used "as a node" in a bigger graph, since we cannot pass messages towards `y`, other than delta distributions. The fact that `y` is observed is not a property of the model, but rather a feature of how we want to use the model (in some inference problem). 

### On `x_prior`
This is really minor but it's a bit odd that you introduce `x_prior` and then copy it into `x_prev`, seemingly because you don't like the variable name `x_prior` in the loop. Better to just not use the variable name `x_prior` (saves a line). I also don't like the explanation of `~` in the comment. I think `~` simply means: `is distributed as`.

### On `for i in 1:n`
This is really low priority, but technically, I don't even think that the iteration `for i in 1:n` should be part of the model specification. I think you just want to specify $p(y_i,x_i | x_{i-1})$ with parameters $x0, A, B, P, Q$ inside the `@model` function. How you _use the model_ (in this case roll out for `n` samples) is determined by the inference task (in this case: infer `p(x|y)`). So, I think it should be written as
```
result = infer(
  model = SSM( x0, A, B, P, Q ),
  data = (y = observations,)
``` 

ie, without the `n` in the `SSM`. Then, when rxinfer executes this inference task, the model gets rolled out over `length(y)`. And I like `infer()` rather than `inference()` because I think it should be our convention to express the action of the function by a verb. It reads better. 

In short, my main feedback is that I think that we don't want to specify **how we use** the model in the model specification phase.

**Comment by bvdmitri** [comment]
Good points, we should consider them when refactoring GraphPPL and creating an actual graph for a model.
